### ISSUE TITLE
Radarr not correctly parsing (and processing) movies it sent to NZBGet

### ISSUE THREAD

**Issue by natoriousbigg** [Radarr/Radarr#1895]
**Description:**

I'm using Radarr 0.2.0.800 with NZBGet. Radarr sometimes find a movie and sends it to NZBGet for download. After the download is complete, however the movie is not recognized and not processed by Radarr. An example is a movie called "Left Behind Vanished: Next Generation (2016)" which Radarr grab an nzb called "Vanished.Left.Behind.Next.Generation.2016.1080p.BluRay.x264-THUGLiNE English". As you can tell, the title is off a bit, although Radarr was able to parse the nzb correctly in the first place from the indexer. But once the download is sent to NZBGet, Radarr no longer recognizes this movie and it isn't showing up under the Queue page. From my understanding of the situation, Radarr doesn't keep track of movies it sent to NZBGet (as opposed to CouchPotato which I knows keeps track of movies it sent to SABnzbd by an nzb ID from my prior experience and presumably NZBGet as well).

**Radarr Version:**

0.2.0.800

**Logs:**

Can generate if needed.

**Comment by hotio** [comment]
So this is a feauture request or what?

**Comment by ladude626** [comment]
I guess it is a feature request. I think there are 3 solution this this issue:

1. List all unparsable/unrecognized downloads is the correct category in NZBGet/SABnzb on the Queue page and allow for manual import.
2. Keep track of all nzbs sent NZBGet/SABnzb by linking them to the associated movies in Radarr's database, and always import them when complete. Since the nzb was already parsed, this may lessen the need to repeatedly parse the download queue every n minutes.
3. For movies already in download, allow an option to use a "lenient" parser algorithm.

**Comment by hotio** [comment]
I think linking the snatched file with an ID, like couchpotato does, would be a great addition. Because I believe the obfuscation is only going to get worse...the downside is, you'll have to trust the indexer with your eyes closed.

**Comment by galli-leo** [comment]
3. Is already there (Settings -> Indexers -> Parser Leniency)

What is the file in question named like? Also what is it's name in your nzbclient?

**Comment by ladude626** [comment]
@galli-leo I would use setting but I don't want to use lenient parsing for Indexers. I want to be able to use that options for files already in the downloaders.

The name in the nzbget is: Vanished.Left.Behind.Next.Generation.2016.1080p.BluRay.x264-THUGLiNE
File inside is: tl-leftbehind.vng-1080p.mkv

Once rename the the nzb in NZBGet to Left.Behind.Vanished.Next.Generation.2016.1080p.BluRay.x264-THUGLiNE it was able to process it.

I do believe using an ID to track downloads is also a good idea.

**Comment by ladude626** [comment]
Here's another example. This movie from Denmark with: https://www.themoviedb.org/movie/425716

English title: Small Town Killers (2017)
Foreign/original title:  Dræberne fra Nibe (2017)

Radarr correctly grab this nzb and sent it to NZBGet: `draeberne.fra.nibe.2017.1080p.bluray.x264-cdn.nzb`
Folder name after download: `cdn-draeberne.fra.nibe.2017.1080p.bluray.x264`
Media file inside: `cdn-draeberne.fra.nibe.2017.1080p.bluray.x264.mkv`

Radarr it is unable to parse this download:

	17-8-21 16:22:35.7|Debug|Parser|Parsing string 'draeberne.fra.nibe.2017.1080p.bluray.x264-cdn'
	17-8-21 16:22:35.7|Debug|Parser|Movie Parsed. draeberne fra nibe - 2017
	17-8-21 16:22:35.7|Debug|Parser|Language parsed: English
	17-8-21 16:22:35.7|Debug|QualityParser|Trying to parse quality for draeberne.fra.nibe.2017.1080p.bluray.x264-cdn
	17-8-21 16:22:35.7|Debug|Parser|Quality parsed: Bluray-1080p v1
	17-8-21 16:22:35.7|Debug|Parser|Release Group parsed: cdn
	17-8-21 16:22:35.7|Debug|DownloadedMovieImportService|Unknown Movie draeberne.fra.nibe.2017.1080p.bluray.x264-cdn

After renaming the NZB to `small.town.killers.2017.1080p.bluray.x264-cdn` in NZBGet, the movie was correctly parsed:

        17-8-21 16:25:05.7|Debug|Parser|Parsing string 'small.town.killers.2017.1080p.bluray.x264-cdn'
	17-8-21 16:25:05.7|Debug|Parser|Movie Parsed. small town killers - 2017
	17-8-21 16:25:05.7|Debug|Parser|Language parsed: English
	17-8-21 16:25:05.7|Debug|QualityParser|Trying to parse quality for small.town.killers.2017.1080p.bluray.x264-cdn
	17-8-21 16:25:05.7|Debug|Parser|Quality parsed: Bluray-1080p v1
	17-8-21 16:25:05.7|Debug|Parser|Release Group parsed: cdn
	17-8-21 16:25:05.7|Debug|DownloadedMovieImportService|draeberne.fra.nibe.2017.1080p.bluray.x264-cdn folder quality: Bluray-1080p v1
	17-8-21 16:25:05.7|Debug|DiskScanService|Scanning '/downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn' for video files
	17-8-21 16:25:05.7|Debug|DiskScanService|1 video files were found in /downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn
	17-8-21 16:25:06.1|Debug|ImportDecisionMaker|Analyzing 1/1 files.
	17-8-21 16:25:06.1|Debug|QualityParser|Trying to parse quality for /downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn/cdn-draeberne.fra.nibe.2017.1080p.bluray.x264.mkv
	17-8-21 16:25:06.1|Debug|QualityParser|Trying to parse quality for cdn-draeberne.fra.nibe.2017.1080p.bluray.x264.mkv
	17-8-21 16:25:06.1|Debug|ImportDecisionMaker|Size: 8206044478
	17-8-21 16:25:06.1|Debug|VideoFileInfoReader|Getting media info from /downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn/cdn-draeberne.fra.nibe.2017.1080p.bluray.x264.mkv
	17-8-21 16:25:06.8|Debug|FreeSpaceSpecification|Skipping free space check when importing
	17-8-21 16:25:06.8|Debug|VideoFileInfoReader|Getting media info from /downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn/cdn-draeberne.fra.nibe.2017.1080p.bluray.x264.mkv
	17-8-21 16:25:06.8|Debug|DetectSample|Runtime is over 90 seconds
	17-8-21 16:25:06.8|Debug|ImportApprovedMovie|Decisions: 1
	17-8-21 16:25:07.0|Debug|MovieFileMovingService|Moving movie file: /downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn/cdn-draeberne.fra.nibe.2017.1080p.bluray.x264.mkv to /movies/foreign/Small Town Killer (2017)/Small.Town.Killers.2017.Bluray-1080p.x264.DTS-cdn.mkv
	17-8-21 16:25:07.0|Debug|DiskTransferService|Move [/downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn/cdn-draeberne.fra.nibe.2017.1080p.bluray.x264.mkv] > [/movies/foreign/Small Town Killer (2017)/Small.Town.Killers.2017.Bluray-1080p.x264.DTS-cdn.mkv]
	17-8-21 16:25:07.1|Info|MovieService|Linking [Small.Town.Killers.2017.Bluray-1080p.x264.DTS-cdn.mkv] > [[tt5458566][Small Town Killers (2017)]]
	17-8-21 16:25:08.1|Debug|DiskScanService|Scanning '/downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn' for video files
	17-8-21 16:25:08.2|Debug|DiskScanService|0 video files were found in /downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn
	17-8-21 16:25:08.2|Warn|DownloadedMovieImportService|RAR file detected, will require manual cleanup
	17-8-21 16:25:08.2|Debug|DownloadEventHub|[small.town.killers.2017.1080p.bluray.x264-cdn] Removing download from NZBGet1 history
	17-8-21 16:25:08.4|Debug|Nzbget|[small.town.killers.2017.1080p.bluray.x264-cdn] Deleting folder '/downloads/movies/draeberne.fra.nibe.2017.1080p.bluray.x264-cdn'.

**Comment by Pseudomax** [comment]
Has this been fixed? I am having the same error...

@natoriousbigg can you explain where you changed the nzb name? Did you do this by downloading the nzb, renaming and them importing into nzbget or did you do this after downloading?

**Comment by Pseudomax** [comment]
No thoughts?

**Comment by fryfrog** [comment]
Sonarr and Radarr are kind of like two black boxes. One of them parses results from indexers and queues things up for download and the other parses the download client. So while the *search* portion may occur via imdbid or tvdbid, the download client parse can *only* see the name of the release. So if it searches for The Movie w/ imdbid 123 and picks one of those results and it turns out the *release* is called "Le Moooveee", it just doesn't recognize it.

In Sonarr v3, you can show unparsed downloads, which makes these easier to deal with. I believe that is in the port to Radarr as v2, but it isn't ready to use yet.

"Fixing" this is not a simple undertaking, I suspect if it ever gets "fixed", that'll happen in Sonarr first and hopefully make its way to Radarr. There are probably some past issues here or on Sonarr's github that have a better explaination of the issue, maybe see if you can find one and link it here?

**Comment by Pseudomax** [comment]
Hi @fryfrog 

Thanks for the response ... much appreciated.

I do understand the naming conventions that are used. Please though can you direct me to the right name convention to use and also where/what to change to get the 'download parse' to recognise things?

Thanks

**Comment by fryfrog** [comment]
Most downloads will be fine, when you hit one that isn't you don't need to go to all the trouble of fixing the name. Just do a manual import and select any criteria it doesn't magically intuit.

**Comment by Pseudomax** [comment]
Hi, sorry to be a little slow on this one ... but I think I have already tried this unsuccessfully...

I have tried:
- manually searching from within Radarr, then manually downloading the NZB. NZBget will accept this and succeed with the download but it is never then imported correctly into the right folder.
- I have done a manually import as well with the NZB but this also succeeds but even after adding the right catagory, Radarr still never finds it and it is never imported to the correct folder.

You are correct though that the ones that fail all have very odd naming of the NZB's when I look at NZGget logs ...
